### PR TITLE
lestarch: updating Ref app to remove phases

### PR DESCRIPTION
**Pull request by LeStarch**
| | |
|:---|:---|
|**_Originating Project/Creator_**| |
|**_Affected Component_**|  |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Builds Without Errors (y/n)_**|  |
|**_Unit Tests Pass (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

This removes the phases setup in ref regarding issues: #1460. This work amounted to several pieces:

1. Correct `polyDb` to not require ctor arguments.
2. Remove unused, old, and outdated files in the Ref topology
3. Create `Topology.hpp` and `Topology.cpp` that
4. Document with comments extensively
5. Rework what goes in `Main.cpp` and what goes in `Topology.cpp`

@ThibFrgsGmz I think this may have accomplished similar lines to #1700...albeit via a different path and for different reasons. I apologize for not catching the overlap between your work in #1700 and the spirit of #1460.

### CODE REVIEW

**Comment by ThibFrgsGmz** [comment]
@LeStarch Indeed, there are similarities but your work is much more complex and useful, no worries!
Once your PR merged, I can rework the "user integration" example in a standalone repo.

File: Ref/Top/Topology.cpp
Comment (timcanham) [inline Ref/Top/Topology.cpp:92]: Minor: Add comment: "... require context arrays. Empty for Reference example."

File: Ref/Top/Topology.hpp
Comment (timcanham) [inline Ref/Top/Topology.hpp:91]: CR?

File: Ref/Top/instances.fpp
Comment (timcanham) [inline Ref/Top/instances.fpp:114]: Since including a header is a break in the typical pattern, add a comment to explain the difference.

File: Svc/ActiveRateGroup/ActiveRateGroup.hpp
Comment (timcanham) [inline Svc/ActiveRateGroup/ActiveRateGroup.hpp:33]: `CHILD` isn't really an F Prime concept; perhaps `CONNECTION_COUNT_MAX`?

**Review by timcanham — CHANGES_REQUESTED** [review]
Minor editorials and changes...

File: RPI/Top/instances.fpp
Comment (LeStarch) [reply RPI/Top/instances.fpp:54]: Yes.  I did not de-phase RPI in this PR.

File: Ref/Top/Topology.hpp
Comment (LeStarch) [reply Ref/Top/Topology.hpp:91]: I don't follow what you mean?

Comment (LeStarch) [reply Ref/Top/Topology.hpp:91]: Carriage return....you wanted a proper file ending.

**Comment by LeStarch** [comment]
@timcanham done.  I am not changing RPI in this PR.

**Comment by LeStarch** [comment]
I am also not working on the code-QL errors as it has flagged items that already exist.

File: RPI/Top/instances.fpp
Comment (timcanham) [reply RPI/Top/instances.fpp:54]: Whoops! Missed that it was PI

**Review by timcanham — APPROVED** [review]

**Comment by bocchino** [comment]
Merging these changes will break the FPP User's Guide, which cites Ref as an example for how to use phases. Is there an alternate example available of using phases somewhere? Maybe in the System Reference?

**Comment by bocchino** [comment]
Also, can we hold off on merging these changes until the FPP User's Guide is updated? I would prefer not to break it.

**Comment by bocchino** [comment]
It looks like the FPP User's Guide can cite the System Reference example:

https://github.com/fprime-community/fprime-system-reference/blob/main/SystemReference/Top/instances.fpp

I created an issue on fprime-community/fpp to track this change and linked this PR.

**Comment by LeStarch** [comment]
@bocchino what is the time frame for this change? I am ok holding off for a bit for the user guide to be updated.

**Comment by bocchino** [comment]
I should be able to turn this around quickly, assuming that the setup of the basic components in the System Reference is similar to the previous Ref example.

**Comment by bocchino** [comment]
I resolved the issue in the FPP docs. This is good to go from my end.

**Comment by LeStarch** [comment]
@bocchino thanks!  I will merge it now.

**Pull request merged**
